pkg/cdclog/buffer.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdclog
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/meta/autoid"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/br/pkg/kv"
+)
+
+// TableBuffer represents the kv buffer of this table.
+// we restore one tableBuffer in one goroutine.
+// this is the concurrent unit of log restore.
+type TableBuffer struct {
+	KvPairs []kv.Row
+	count   int
+	size    int64
+
+	KvEncoderFn func(autoid.Allocators, table.Table) (kv.Encoder, error)
```

Comment:
eh. settable member for lazy create and maybe for mock in the future? I can change it to private member.

Reply:
Keeping KvEncoder public should be enough for mocking since you could fill in a non-nil mock instance, unless you do want to test the lazy init behavior.